src/simpleNbox.cpp

```diff
@@ -143,16 +184,33 @@ void SimpleNbox::setData( const std::string &varName,
             set_c0(data.getUnitval(U_PPMV_CO2).value(U_PPMV_CO2));
         }
         else if( varNameParsed == D_VEGC ) {
-            H_ASSERT( data.date == Core::undefinedIndex(), "date not allowed" );
+            // For `veg_c`, `detritus_c`, and `soil_c`, if date is not
+            // provided, set only the "current" model pool, without
+            // touching the time series variable. This is to
+            // accommodate the way the INI file is parsed. For
+            // interactive use, you will usually want to pass the date
+            // -- otherwise, the current value will be overridden by a
+            // `reset` (which includes code like `veg_c = veg_c_tv.get(t)`).
+            //
+            // TODO: Should we create special variables `veg_c0`,
+            // `soil_c0`, etc. that are set in the INI file and
+            // require a date for `veg_c`, `soil_c`, etc.?
```

Comment:
I.e. matching the approach taken with NPP?

Reply:
Exactly. I clarified this comment in 55c9341f43ef24026de736c22dd66db8fbc390db.

Reply:
Conversation moved to #318.